wagtail/snippets/views/snippets.py

```diff
@@ -903,3 +934,11 @@ def get_urlpatterns(self):
         ]
 
         return urlpatterns + legacy_redirects
+
+    def on_register(self):
+        super().on_register()
+
+        url_finder_class = type(
+            "_SnippetAdminURLFinder", (SnippetAdminURLFinder,), {"model": self.model}
+        )
+        register_admin_url_finder(self.model, url_finder_class)
```

Comment:
Moving this here instead of in `register_snippet()` as I think it fits better here? The `ModelViewSet` also does a similar thing:

https://github.com/wagtail/wagtail/blob/ce43564bdebe45115d2aab14b35370a64cb7a313/wagtail/admin/viewsets/model.py#L114-L124

Reply:
Yep, I think this makes sense, thanks!